From ca96b74855d5693cc5c82f569082fbefc255711a Mon Sep 17 00:00:00 2001
From: Siddharth Manthan <siddharth_manthan@outlook.com>
Date: Fri, 7 May 2021 22:19:57 +0530
Subject: [PATCH 18/24] arm64: dts: qcom: msm8916-samsung-gt58: Add Vibrator

Motor is powered by VPH_PWR (aka Vbat) and controlled by GPIO.

Signed-off-by: Siddharth Manthan <siddharth_manthan@outlook.com>
---
 .../boot/dts/qcom/msm8916-samsung-gt58.dts     | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt58.dts b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt58.dts
index 384b396b00..79569ea4a2 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt58.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt58.dts
@@ -7,4 +7,22 @@
 / {
 	model = "Samsung Galaxy Tab A 8.0 (2015)";
 	compatible = "samsung,gt58", "qcom,msm8916";
+
+	vibrator {
+		compatible = "gpio-vibrator";
+		enable-gpios = <&msmgpio 76 GPIO_ACTIVE_HIGH>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&vibrator_en_default>;
+	};
+};
+
+&msmgpio {
+	vibrator_en_default: vibrator-en-default {
+		pins = "gpio76";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
 };
-- 
2.25.1

